src/components/popover-container/popover-container.component.tsx

```diff
@@ -14,33 +14,123 @@ import Icon from "../icon";
 import createGuid from "../../__internal__/utils/helpers/guid";
 import { filterStyledSystemPaddingProps } from "../../style/utils";
 
-const paddingPropTypes = filterStyledSystemPaddingProps(
-  styledSystemPropTypes.space
+export interface RenderOpenProps {
+  tabIndex: number;
+  isOpen?: boolean;
+  "data-element": string;
+  onClick: (ev: React.MouseEvent<HTMLElement>) => void;
+  ref: React.RefObject<HTMLButtonElement>;
+  "aria-label"?: string;
+  id?: string;
+}
+
+const renderOpen = ({
+  tabIndex,
+  onClick,
+  "data-element": dataElement,
+  ref,
+  "aria-label": ariaLabel,
+  id,
+}: RenderOpenProps) => (
+  <PopoverContainerOpenIcon
+    tabIndex={tabIndex}
+    onAction={onClick}
+    data-element={dataElement}
+    ref={ref}
+    aria-label={ariaLabel}
+    aria-haspopup
+    id={id}
+  >
+    <Icon type="settings" />
+  </PopoverContainerOpenIcon>
+);
+
+export interface RenderCloseProps {
+  "data-element": string;
+  tabIndex: number;
+  onClick: (ev: React.MouseEvent<HTMLElement>) => void;
+  ref: React.RefObject<HTMLButtonElement>;
+  "aria-label": string;
+}
+
+const renderClose = ({
+  "data-element": dataElement,
+  tabIndex,
+  onClick,
+  ref,
+  "aria-label": ariaLabel,
+}: RenderCloseProps) => (
+  <PopoverContainerCloseIcon
+    data-element={dataElement}
+    tabIndex={tabIndex}
+    onAction={onClick}
+    ref={ref}
+    aria-label={ariaLabel}
+  >
+    <Icon type="close" />
+  </PopoverContainerCloseIcon>
 );
 
-const PopoverContainer = ({
+export interface PopoverContainerProps extends PaddingProps {
+  /** A function that will render the open component
+   *
+   * `({tabIndex, isOpen, data-element, onClick, ref, aria-label}) => ()`
```

Comment:
do we still need that information here?

Reply:
Probably not, but it's actually more readable in storybook thanks to this comment.